src/api/common_api.js

```diff
@@ -138,6 +142,126 @@ module.exports = {
             }
         },
 
+        // bucket lifecycle
+
+        bucket_lifecycle_rule_expiration: {
+            type: 'object',
+            properties: {
+                days: {
+                    minimum: 1,
+                    type: 'integer'
+                },
+                date: {
+                    idate: true
+                },
+                expired_object_delete_marker: {
+                    type: 'boolean'
+                }
+            }
+        },
+        bucket_lifecycle_rule_status: {
+            enum: ['Enabled', 'Disabled'],
+            type: 'string'
+        },
+        bucket_lifecycle_rule_and_operator: {
+            type: 'object',
+            properties: {
+                prefix: {
+                    type: 'string'
+                },
+                tags: {
+                    type: 'array',
+                    items: {
+                        $ref: '#/definitions/tag',
+                    },
+                },
+                object_size_greater_than: {
+                    type: 'number',
+                },
+                object_size_less_than: {
+                    type: 'number',
+                },
+            },
+        },
+        bucket_lifecycle_rule_filter: {
+            type: 'object',
+            properties: {
+                prefix: {
+                    type: 'string'
+                },
+                tag: {
```

Comment:
just to make my previous comment clearer - I think `tags` should be an array, and we can remove the `and` property. we should always treat any existing property as part of an `And`. unless it conflicts with a different possible behavior that I'm missing

Reply:
how about [get_bucket_lifecycle()](https://github.com/noobaa/noobaa-core/pull/6901/files/8c02125d28707ce2371c7378fe5b84d6dc0638ab#diff-431c35029c5e0154ce478a1f6672c076db12f50276241c058eb26cf4db63b0ab)? Should we always return `And`, even if `And` actually was not used during `put_bucket_lifecycle()`?

Reply:
This is an intelligent bright idea of generalization, however simplifying internal data representation beyond the point of not being able to reconstruct the original filter, is not a good idea IMHO.

Reply:
Maybe I'm missing something here? And doesn't just mean we have more than one filter? like one prefix and tag or more than one tags? why can't we understand that from what we have in the DB?

Reply:
@jackyalbo, Alex's main concern is that the user can set a policy with `And` and only one element. in that case, when the user gets the policy, a different structure is returned. 
@baum correct me if I'm wrong